src/iosrv/ult.c

```diff
@@ -452,8 +452,18 @@ dss_ult_create(void (*func)(void *), void *arg, int ult_type, int tgt_idx,
 free:
 	if (attr != ABT_THREAD_ATTR_NULL) {
 		rc1 = ABT_thread_attr_free(&attr);
-		if (rc == ABT_SUCCESS)
-			rc = rc1;
+		if (rc1 != ABT_SUCCESS)
+			/* The child ULT has already been created,
+			 * we should not return the error for the
+			 * ABT_thread_attr_free() failure; otherwise,
+			 * the caller will free the parameters ("arg")
+			 * that is being used by the child ULT.
+			 *
+			 * So let's ignore the failure, the worse case
+			 * is that we may leak some DRAM.
+			 */
+			D_ERROR("ABT_thread_attr_free failed: %d\n",
+				dss_abterr2der(rc1));
```

Comment:
Probably we should just ignore the return code from a 'free' call (I mean above changes & 'rc1' could be removed).

BTW, the attr is only created/freed when the @stack_size isn't 0, so I don't think it's the root cause of DAOS-4244.

Reply:
> Probably we should just ignore the return code from a 'free' call (I mean above changes & 'rc1' could be removed).
> 

Checking the ABT_thread_attr_free() return value will help us to understand what happened even though we do not forward it to the caller.

> BTW, the attr is only created/freed when the @stack_size isn't 0, so I don't think it's the root cause of DAOS-4244.
>

I agree that it may be not the root reason for this ticket, but it is potential general trouble for creating ULT use cases. So it is better to fix it. I have another patch (PR#2019) to fix DAOS-4244 issue.